meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in

```diff
@@ -1,21 +1,25 @@
 # common boot handlers
 setenv fio_msg "FIO:"
+setenv ostree_split_boot "@@OSTREE_SPLIT_BOOT@@"
+setenv ostree_deploy_usr "@@OSTREE_DEPLOY_USR_OSTREE_BOOT@@"
 
 # uEnv handling
-setenv bootcmd_resetvars 'setenv kernel_image; setenv bootargs; setenv kernel_image2; setenv bootargs2'
-setenv bootcmd_otenv 'run bootcmd_resetvars; ext4load ${devtype} ${devnum}:${rootpart} ${loadaddr} /boot/loader/uEnv.txt; env import -t ${loadaddr} ${filesize} kernel_image bootargs kernel_image2 bootargs2'
-setenv bootcmd_bootenv 'setenv bootfirmware_version; ext4load ${devtype} ${devnum}:${rootpart} ${loadaddr} ${ostree_root}/usr/lib/firmware/version.txt; env import -t ${loadaddr} ${filesize} bootfirmware_version'
+setenv bootcmd_resetvars 'setenv kernel_image; setenv bootdir; setenv bootargs; setenv kernel_image2; setenv bootdir2; setenv bootargs2'
+setenv bootcmd_ostree_bootpart 'if test "${ostree_split_boot}" = "1"; then setenv ostree_bootpart "${bootpart}"; else setenv ostree_bootpart "${rootpart}"; fi'
+setenv bootcmd_otenv 'run bootcmd_resetvars; ext4load ${devtype} ${devnum}:${ostree_bootpart} ${loadaddr} /boot/loader/uEnv.txt; env import -t ${loadaddr} ${filesize} kernel_image bootdir bootargs kernel_image2 bootdir2 bootargs2'
+setenv bootcmd_bootenv 'setenv bootfirmware_version; if test "${ostree_deploy_usr}" = "1"; then ext4load ${devtype} ${devnum}:${ostree_bootpart} ${loadaddr} ${bootdir}/version.txt; else ext4load ${devtype} ${devnum}:${rootpart} ${loadaddr} ${ostree_root}/usr/lib/firmware/version.txt; fi; env import -t ${loadaddr} ${filesize} bootfirmware_version'
```

Comment:
@ricardosalveti iiuc, ak-lite/ostree besides adjusting uEnv.txt also copies boot images (etc SPL, u-boot.itb)/version.txt file, and path to both copies are saved in `bootdir` and `bootdir2` vars in uEnv.txt

The only question is why do we need two dedicated variables `ostree_split_boot` and `ostree_deploy_usr`?

Reply:
If `ostree_split_boot == 1` (which means that LUKS/dm-crypt is enabled), but `ostree_deploy_usr != 1` will this part after `else` make sense?:
`ext4load ${devtype} ${devnum}:${rootpart} ${loadaddr} ${ostree_root}/usr/lib/firmware/version.txt`

Reply:
Split boot is simply to separate the ostree /boot content in a separated partition, and also used on UEFI-based systems (e.g. intel). With split the user can decide to encrypt or not the rootfs, as both cases should be supported.

Deploy usr is to force ostree to deploy files under rootfs/usr/lib/ostree-boot (can be anything) to the first partition, which is important for encrypted rootfs.